crates/pixi_command_dispatcher/tests/integration/main.rs

```diff
@@ -363,3 +368,42 @@ pub async fn test_stale_host_dependency_triggers_rebuild() {
         "Expected only package-a and package-b to be rebuilt"
     );
 }
+
+#[tokio::test]
+pub async fn instantiate_backend_with_from_source() {
+    let backend_name = PackageName::new_unchecked("some-source-backend");
+
+    // Create a temporary directory and build the fake backend package
+    let tempdir = tempfile::tempdir().unwrap();
+    let package_dir = SourceBackendBuilder::new(
+        "some-source-backend",
+        source_backend::BackendType::PixiBuildPython,
+    )
+    .build(tempdir.path())
+    .unwrap();
+
+    let dispatcher = CommandDispatcher::builder()
+        .with_cache_dirs(default_cache_dirs())
+        .with_executor(Executor::Serial)
+        .with_backend_overrides(BackendOverride::InMemory(
+            InMemoryOverriddenBackends::Specified(HashMap::from_iter([(
+                backend_name.as_source().to_string(),
+                PassthroughBackend::instantiator().into(),
+            )])),
+        ))
+        .finish();
+
+    // Use PixiSpec::Path to test path-based resolution and installation
+    dispatcher
+        .instantiate_tool_environment(InstantiateToolEnvironmentSpec::new(
+            backend_name,
+            PixiSpec::Path(PathSpec {
+                path: Utf8TypedPathBuf::from(package_dir.to_str().unwrap()),
+            }),
+            Vec::from([Url::from_str("https://prefix.dev/conda-forge")
+                .unwrap()
+                .into()]),
+        ))
+        .await
+        .unwrap();
```

Comment:
I think it would be good to validate that it also rebuilds, this seems to work locally. 

Also is there any way we can make this faster? This is a 25 second test.